nilearn/glm/first_level/first_level.py

```diff
@@ -959,107 +949,403 @@ def first_level_from_bids(dataset_path, task_label, space_label=None,
+                f"Entity {filter_[0]} for {filter_} is not a possible filter. "
+                f"Only {supported_filters} are allowed.")        
+
+        if (not isinstance(filter_[1], str) 
+            or not _is_alphanumeric(filter_[1])):
+            raise TypeError('BIDS labels in img_filters must be alphanumeric. '
+                            f"Got {type(filter_[1])} for {filter_} instead.")
+
+
+def _is_alphanumeric(string: str) -> bool:
+    return all(char.isalnum() for char in string)
+
+def _filter_for_bids_query(task_label: str,
+                           space_label: str | None = None,
+                           supported_filters: list[str] | None = None,
+                           extra_filter: list[tuple[str, str]] | None = None
+                           ) -> list[tuple[str, str]]:
+    """Return a filter to specific files from a BIDS dataset.
+
+    Parameters
+    ----------
+    task_label : :obj:`str`
+        Task_label as specified in the file names like _task-<task_label>_.
+
+    space_label : :obj:`str` or None, optional
+        Specifies the space label of the preprocessed bold.nii images.
+        As they are specified in the file names like _space-<space_label>_.
+
+    supported_filters : :obj:`list` of :obj:`str` or None, optional
+        List of authorized BIDS entities
+
+    extra_filter : :obj:`list` of :obj:`tuple` (str, str) or None, optional
+        _description_
+
+    Returns
+    -------
+    Filter to be used by :func:`get_bids_files`: \
+        :obj:`list` of :obj:`tuple` (str, str)
+        filters
+
+    """
+    filters = [('task', task_label)]
+
+    if space_label is not None:
+        filters.append(('space', space_label))
+
+    if extra_filter and supported_filters:
+        for filter_ in extra_filter:
+
+            if filter_[0] not in supported_filters:
+                warn(f"The filter {filter_} will be skipped. "
+                     f"'{filter_[0]}' is not among the supported filters. "
+                     f"Allowed filters include: {supported_filters}")
+                continue
+
+            filters.append(filter_)
+
+    return filters
+
+
+def _check_bids_image_list(imgs: list[str] | None,
+                           sub_label: str,
+                           filters: list[tuple[str, str]]) -> None:
+    """Check input BIDS images.
+
+    Check that:
+        - some images were found
+        - if more than one image was found, check that there is not more than
+          one image for a given session / run combination.
+
+    Parameters
+    ----------
+    imgs : list[str] or None
+        List of image fullpath filenames.
+
+    sub_label : :obj:`str`
+        Subject label as specified in the file names like _sub-<sub_label>_.
+
+    filters : :obj:`list` of :obj:`tuple` (str, str)
+        Filters of the form (field, label) used to select the files.
+        See :func:`get_bids_files`.
+
+    """
+    if not imgs:
+        raise ValueError('No bold files found '
```

Comment:
```suggestion
        raise ValueError('No BOLD files found '
```
Since bold is an acronym.